app/models/user.rb

```diff
@@ -0,0 +1,25 @@
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  property :id,       Serial
+  property :name,     String 
+  property :email,    String
+  property :password_digest, Text
+
+  def password=(password)
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(name, password)
+    user = self.first(:name => name)
```

Comment:
you don't need self.first here, you can just call first as self is implicit in ruby